### PR TITLE
Add Kernel#lambda proc without block warning

### PR DESCRIPTION
**Pull request by bjfish**

### CODE REVIEW

File: src/main/java/org/truffleruby/core/kernel/KernelNodes.java [hunk 1]
```diff
@@ -992,6 +998,15 @@ private boolean isLiteralBlock(Node callNode) {
             return false;
         }
 
+        private void warnProcWithoutBlock() {
+            if (warnNode == null) {
+                CompilerDirectives.transferToInterpreterAndInvalidate();
+                warnNode = insert(new WarnNode());
+            }
+            final SourceSection sourceSection = getContext().getCallStack().getCallerFrameIgnoringSend().getCallNode().getEncapsulatingSourceSection();
```
Comment (eregon) [inline src/main/java/org/truffleruby/core/kernel/KernelNodes.java]: This is fairly long, could we have a helper like CallStackManager#getTopMostUserSourceSection(), like getTopMostUserCallNode() but returning a SourceSection?

Comment (bjfish) [reply src/main/java/org/truffleruby/core/kernel/KernelNodes.java]: Added, please review.

File: src/main/java/org/truffleruby/language/CallStackManager.java [hunk 2]
```diff
@@ -176,6 +176,11 @@ public Node getTopMostUserCallNode() {
         });
     }
 
+    @TruffleBoundary
+    public SourceSection getTopMostUserSourceSection() {
+        return getTopMostUserCallNode().getEncapsulatingSourceSection();
```
Comment (eregon) [inline src/main/java/org/truffleruby/language/CallStackManager.java]: What if `getTopMostUserCallNode()` returns `null`? I think `getTopMostUserSourceSection` should just return `null` as well in that case.

Comment (bjfish) [reply src/main/java/org/truffleruby/language/CallStackManager.java]: Fixed, thanks.

**Pull request merged**